### DISCUSSION TITLE
Magazines

### DISCUSSION THREAD

**Discussion by nguillaumin** [atari-legend/atari-legend#188] [Show and tell] [answered]
Hi all, @stgraveyard,

I've been working on adding back a magazine section to AL. There's a new "Mags" entry in the menu showing all the magazine in the database. Choosing a magazine shows all the issues that we have with an index for each, listing the game reviews and other articles.

![mag1](https://user-images.githubusercontent.com/762512/185797272-00a4a273-cd48-439d-b7be-d61db4db1fe4.png)

There's also a new magazine box on the games page:

![mag2](https://user-images.githubusercontent.com/762512/185797312-c007816c-e79b-4334-baa5-edb5ad7d51cc.png)

I chose to not store the magazines on AL but rather point to Archive.org . This way we don't have to store them and they have a decent reading interface that we don't have to implement ourselves. For example: https://archive.org/details/Atari_ST_User_Issue_102_1994-07_Europress_GB/?view=theater . Anyone can upload files to Archive.org so if you have some mags that are not there yet, it's easy to upload.

The little magazine icons on the site are pointing to Archive.org in reader mode. Unfortunately I was not able to open a specific page as the page numbers are inconsistent across all the magazines (it depends how it was scanned and uploaded on Archive.org).

There was already some mags content in the database (dating from an old time I assume!). I completed Atari ST User 102 as an example: https://dev.atarilegend.com/magazines/4

It's deployed on dev, let me know what you think?

**Comment by stgraveyard** [comment, marked as answer]
Nico, 

This is amazing. I think it looks good and responsiveness seems to be working. Also the fact that the mags are not stored at the server is perfect, we don't need to reinvent the wheel.
But I need some info. The covers used on AL, are these in the DB? How does the indexing work? I see for STF you have a list of the content, but no pages, for ATARI ST USER you have the pages and the link to the magazine. Is this all manual database work, or are these pages available somewhere.

Suggestion, is it maybe also, in someway possible to do a search for a word or game or program in the magazines. I know this is possible in archive.net. Can we in some way link this search functionality to AL? 

Amazing work Nico, something I have wanted for a long time.

Thanks a lot,
Maarten

**Reply by nguillaumin** [reply]
The covers are stored on the site yes, but just the cover. It can be uploaded in CPANEL, and there's also a button to automatically fetch them from Archive.org when adding a mag that's stored on Archive.org (it's possible to add mags without an Archive.org link, but it means obviously there will be no link to read them). I advise checking out the magazine section in CPANEL: https://dev.atarilegend.com/admin/magazines/magazines

The indexing is manual DB work unfortunately. For Atari ST User I just put all the index myself as an example. All the other ones were already in the DB (legacy data?) and some of them had some links to games, some of them were empty. Unfortunately there's no way to generate these indexes automatically, they are not available on Archive.org.

Search inside a magazine is not really possible with the current approach of not storing the content in the database. Even so it would we too much work :wink:

**Reply by stgraveyard** [reply]
Ok, as always, the CPANEL part is concise and crystal clear. I understand now. It is a thourough system you have set up. A hell of a lot of DB work, but definitely worth it. I have a whole batch of magazines myself, I guess I will index a mag from time to time until we have done them all...It is great Nico.

How do you suggest we continue? Perhaps you can add a button like you did for reviews and articles, which, if active, keeps the mag or set of mags offline, until we have completed the indexing?

**Reply by nguillaumin** [reply]
I'll be on holidays next week and the week after without a computer, so I will look at deploying that to prod when I'm back.

I would say we don't need a "draft" mode here. Mags will appear on the site as they are put in the DB, especially since it's quick-ish to add a mag issue.

As a preparation step we could at least check that all the mags that are already in the DB are on archive.org, and if not upload them. This way we will at least get the covers and reading links.

**Reply by stgraveyard** [reply]
Ok sounds good. Have a great holiday and we will pick this up than when you are back? 

I do have something to perhaps thing about Nico. It all depends on how complex we will make this section of course but, here is a shot of the cpanel : 

![image](https://user-images.githubusercontent.com/4339294/186273669-9cb0fed6-a203-412d-968c-2063b0b53586.png)

We can link to a game, we can link to a program (I did'nt even know we had regular software in the DB?), but shouldn't we also be able to link to a person? An individual? If we are dealing with an interview perhaps? Or a making of? 
And what if we have a behind the scenes article about a game together with its creator, could we have 2 links to that page? first one article --> game, second one article --> individual? Something like that. 

Also, would it be possible to add a search on the title field? Just a thought. Are those the actual titles of the articles, news pages, sections from the magazines? Or is this a field where we can add search criteria? Or could we add a suplimental field where we can add search criteria which the user can search for?

These things just popped in my head today. 

I don't want to go overboard ofcourse...

**Reply by nguillaumin** [reply]
- Programs: These are the software that are on menu disks. So in fact we do have software in the DB now because of this :wink: 
- Link to individuals: Yes I thought of the same thing while I was making the example with Atari ST User 102, there's an interview of Jeff Minter I think. It would make sense, I will try to implement
- If an article is for multiple things at once, not sure. I think associating an index with a game + individual would be ok. The problem is if there are multiple individuals or games in the same article. We could allow multiple entries per rows, but I'm not sure how the CPANEL would work for that... But there will always be the option to create multiple rows pointing to the same page like you said
- Titles: These are the article titles as in the magazine. For example in a mag the reader's mail could have the title "Mailbag". Or if there's a review of Hunter, there could be a specific title to the article like "Become a spy!", etc. When it's a simple review of a game without any specific title it can be left blank
- Search: We could make title searchable, but I don't really see the use case. If we connect index to games, software and individuals, then you can search for the game (or individual) and get all the articles for it from the game page. I don't see why people would search specifically for article titles?

**Reply by stgraveyard** [reply]
Yes, I agree with you on all points. About the last point, we can indeed search for individual or game, it has to start with the game page always, but I guess it is ok. 

I would really like to see an interview linked to an individual and how it is presented on the game page. Maybe when you are back from holiday? If that is done, we can go to prod and I gladly (and slowly) start indexing my ST Format collection :-)

**Reply by nguillaumin** [reply]
I started looking into this, but I'm not sure how to display the interviews of an individual. On the game page we display the individuals in the " game info"  tile. I don't think it would be the right place and there would not be enough space anyway (considering there may be many interviews of a single individual, across all the magazines).

![1](https://user-images.githubusercontent.com/762512/189483037-77a02ad0-eb34-4b41-9a14-19ac2c0aa64d.png)

The other option would be to add them to the new "magazine" tile, but again there may be many interviews across all magazines so it will result in a long list, which in addition has nothing to do with the game itself (it will just be arbitrary interview of the individuals of the game).

So I'm not sure how to present it. Maybe we would need a new page for individuals, where we could list all their games, magazine interviews, AL interviews, etc...

Do you have any suggestion?

**Reply by stgraveyard** [reply]
Nico, 

I think when we go to the trouble of manual indexing, we index as much as possible. This is something I always wanted myself, and something that is not present anywhere else. For me, I do think the magazine box would be perfectly suited for this : 

![image](https://user-images.githubusercontent.com/4339294/189981748-67eba29b-e5e0-4f7f-a4ca-dc74ea8d3979.png)

We could add an extra little piece of info like 'review', 'interview', 'article'...

We could also enhance the actual magazine page, with a search bar : 

![image](https://user-images.githubusercontent.com/4339294/189982373-6b1b3f9c-e8c2-473a-af8e-7026893b7277.png)

So people can search for game reviews, articles, interviews and so on in the magazines we indexed.

my 2 cents.

**Reply by nguillaumin** [reply]
The CPANEL allows this indexing already. The problem for me is using the magazine box. I made an example to explain. Let's say you're on the game page for Llamatron made by Jeff Minter. If we were listing all the magazines where there's an interview of Jeff Minter it would look something like this:

![Rock-n-Roll-Clams-Atari-ST-game-Atari-Legend](https://user-images.githubusercontent.com/762512/189992561-4f6c15c7-fb68-4815-a45d-750522edb6af.png)

I only put 9 in my example here but there may be many more, for popular individuals that have been interviewed many times. I think that will look ugly, plus it's misleading because these interviews have nothing to do with the game (Llamatron). If you click on a magazine link you will see an interview that is completely unrelated to Llamatron...

**Reply by MugUK** [reply]
Just include the game review(s) for the game. 

Add the interviews to the person(s) being interviewed (even if it's not about a game).

Would that split the above box and not take up as much space?

**Reply by nguillaumin** [reply]
> Add the interviews to the person(s) being interviewed (even if it's not about a game).

This is what will cause my example, if we list in the magazine card any interview for any individual that is involved in a game.

**Reply by nguillaumin** [reply]
To be clear I agree we should index interviews in chanel, I'm just not l convinced we should list them in the magazine card of a game because they are unrelated to the game. I would just list the game reviews there for now.

**Reply by stgraveyard** [reply]
OR ... Nico, what is you can 'collapse' the tile. Like eg give priority to review and articles and up to 10 lines. If there are more, you can press the 'more' link and the tile grows bigger? Wouldn't that solve it all? 

I understand the problem with the interview not having anything to do with the game, but still, with a collapse it doesn't matter that much... ?

**Reply by nguillaumin** [reply]
For now I have deployed without this. We have to put some magazine interviews in the database first, then we'll see how we can display them.

**Reply by stgraveyard** [reply]
Very cool. Nico, I will do anouncement on facebook and twitter this week. But I was wondering, should I perhaps ask in the community if someone is interested in helping with all the indexing?

**Reply by stgraveyard** [reply]
@Brume-AL @MugUK  what do you think about this?

**Reply by MugUK** [reply]
Fine by me. We'll have to sort out some kind of group effort to tackle either a publisher of magazine x at a time and get all the reviews/interviews into the db. Would this be easier via a bulk import of a CSV file, or would it be done manually using the new/old Cpanel?

**Reply by nguillaumin** [reply]
I've tried to make it easy & fast in cpanel so that we can use it, rather than rely on imports. Imports would be cumbersome I think to be able to match specific games or individuals just based on the name. But I'm always open to feedback if you see ways to improve.

For splitting the work, for now I'm focusing on the french ST Magazine. That will take me a while as there are over 100 issues I think...

**Reply by stgraveyard** [reply]
Richard Davey is now on board. I have made him admin, he wants to help fill the mag section!

**Reply by stgraveyard** [reply]
Nico, would be nice perhaps if we can also upload the coverdiscs?

**Reply by nguillaumin** [reply]
Yeah I thought of that. I may add it later.

**Comment by MugUK** [comment]
Looks fantastic. Great work Nico. 

I would prefer the link to open the magazine in a new tab purely because I like having AL open as well  :)

**Reply by nguillaumin** [reply]
Thanks! All across the site I have not specified opening new tabs because I prefer the user to be able to choose (e.g. CTRL+click or 3rd mouse button). For example the YouTube videos also open in the same tab by default. So I'd prefer to keep it that way if possible.

**Reply by stgraveyard** [reply]
I totally understand Mike here, as it also bugged me while testing. I open a magazine, close it and AL is gone. But then again, I had no idea about the ctrl+click option. And I do follow Nico as well to keep everything uniform.

**Comment by Brume-AL** [comment]
This looks very cool 👍👍👍
Have a suggestion: would it be possible to add a flag with the language used somewhere? I mean, will we provide magazines in English, or will we add French, German..., too?

**Reply by nguillaumin** [reply]
You can already set the country of origin when creating a magazine in CPANEL , which is then displayed in the magazine list on the main site. That implies the language as well, although we could add a separate language field later if needed.
